README.md

```diff
@@ -1,204 +1,151 @@
+* `children`: A list of `DynamicComponent` elements that will be added to this component as children (only supported by `ElementGroup` and `RadioGroupButton` as of now).
 
 **DynamicProperty**
 
-O DynamicProperty é referente à propriedade que irá adicionada no componente
+This entity represents a command that will be used to style and configure a `DynamicComponent` and has the following attributes:
 
-* `name`: Nome da propriedade a ser adicionada no componente
-* `type`: Tipo da propriedade (color, dimen, string)
-* `value`: O valor da propriedade
+* `name`: Name that will be associated to the command to be applied to the component
+* `type`: Property type (e.g color, dimen, string, integer, float)
+* `value`: The value associated to this property
 
-**Exemplos de JSON**
+You can see some example custom components we've built by taking a look at this project's modules,
+specifically:
+
+- [Picasso][1]
+- [Glide][2]
+- [Canarinho][3]
+- [Lottie][4]
 
-* Componente de Text (TextView) com texto "Teste componente text", textColor #333839 e textSize de 25 e sem childs
-```
-{
-  "type": "text",
-  "properties": [
-    {
-      "name": "text",
-      "type": "string",
-      "value": "Teste componente text"
-    },
-    {
-      "name": "textColor",
-      "type": "color",
-      "value": "#333839"
-    },
-    {
-      "name": "textSize",
-      "type": "dimen",
-      "value": "25"
-    }
-  ],
-  "children": null
-}
-```
 
-* Componente elementGroup(LinearLayout) com orientation horizontal e botões como children do elementGroup assim alinhando um botão do lado do outro
+**Example usage**
+
+The JSON
+
+```json
+[
+  {
+    "type": "text",
+    "properties": [
+      {
+        "name": "text",
+        "type": "string",
+        "value": "Hello, Android!"
+      },
+      {
+        "name": "textColor",
+        "type": "color",
+        "value": "#4A4A4A"
+      },
+      {
+        "name": "textSize",
+        "type": "dimen",
+        "value": "30"
+      }
+    ],
+    "children": null
+  }
+]
 ```
-{
-  "type": "elementGroup",
-  "properties": [
-    {
-      "name": "orientation",
-      "type": "string",
-      "value": "horizontal"
-    }
-  ],
-  "children": [
-    {
-      "type": "button",
-      "properties": [
-        {
-          "name": "text",
-          "type": "string",
-          "value": "Não"
-        }
-      ],
-      "children": null
-    },
-    {
-      "type": "button",
-      "properties": [
-        {
-          "name": "text",
-          "type": "string",
-          "value": "Sim"
-        }
-      ],
-      "children": null
-    }
-  ]
-}
+
+and Kotlin code
+
+```kotlin
+
+val parent = findViewById<FrameLayout>(R.id.parent)
+
+val json = MainActivity::class.java
+    .getResource("/assets/example_simple.json")
+    .readText()
+
+val creator = DynamicViewCreator.Builder()
+    .build()
+
+creator.createViewFromJson(parent, json, this)

```

Comment:
How about add some comments about the code here? Like "The json that you want to use" or "This is the view creator, you can use the default builder or add your own components".
The code does not mention the action listener, just passing "this" as a parameter

Reply:
I think there should be a section talking about the action listener, but not redundant comments like "this is the json you want to use". There's also a section about custom components so I don't think the other comment brings value either. IMHO.

Reply:
Sure, i also think that this section should exists. But in this context the README is just passing "this" as a parameter. Maybe the section should be before this code and show that the activity is implementing the listener

Reply:
Yup, agreed. Will work on that.